Library/Homebrew/settings.rb

```diff
@@ -0,0 +1,39 @@
+# typed: true
+# frozen_string_literal: true
+
+# Helper functions for reading and writing settings.
+#
+# @api private
+module Settings
+  extend T::Sig
+
+  module_function
+
+  sig { params(setting: T.any(String, Symbol), repo: Pathname).returns(T.nilable(String)) }
+  def read(setting, repo: HOMEBREW_REPOSITORY)
+    return unless (repo/".git/config").exist?
+
+    repo.cd do
+      Utils.popen_read("git", "config", "--get", "homebrew.#{setting}").chomp.presence
+    end
+  end
+
+  sig { params(setting: T.any(String, Symbol), value: T.any(String, T::Boolean), repo: Pathname).void }
+  def write(setting, value, repo: HOMEBREW_REPOSITORY)
+    return unless (repo/".git/config").exist?
+
+    repo.cd do
+      T.unsafe(self).safe_system "git", "config", "--replace-all", "homebrew.#{setting}", value.to_s
```

Comment:
Can `extend` or something be used here to avoid the gross `T.unsafe(self).`? CC @reitermarkus

Reply:
Should work without if the module is moved into the `Homebrew` namespace.

Reply:
Is this something that, ideally, should be done for all modules?

Reply:
Also, doing this means I no longer need `require "completions"`, right?

Reply:
> Is this something that, ideally, should be done for all modules?

Yes. That in addition to having all paths match the contained module names.

Reply:
> I'm guessing no. It will probably slow everything down if too much stuff is loaded each time, right?

Being in the `Homebrew` namespace has nothing to do with whether it is loaded.

Reply:
I moved each module to `Homebrew` like this:

```ruby
module Homebrew
  module Completions
```

Is this what you meant?

I'm still getting this `brew typecheck` issue, though:

```console
$ brew typecheck
completions.rb:59: Method ohai does not exist on Homebrew::Completions https://srb.help/7003
    59 |      ohai "Homebrew completions for external commands are unlinked by default!"
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Note:
    Did you mean to include Kernel in this module?
  Did you mean:
    utils.rb:110: Kernel#ohai
     110 |  def ohai(title, *sput)
            ^^^^^^^^^^^^^^^^^^^^^^
```

I get this for `ohai`, `puts`, and `safe_system`.

Reply:
You'll need a `completions.rbi` file with `include Kernel`.

Reply:
Thanks!